totalRP3/modules/map/MapScannersManager.lua

```diff
@@ -0,0 +1,101 @@
+----------------------------------------------------------------------------------
+--- Total RP 3
+---	---------------------------------------------------------------------------
+--- Copyright 2018 Renaud "Ellypse" Parize <ellypse@totalrp3.info> @EllypseCelwe
+---
+---	Licensed under the Apache License, Version 2.0 (the "License");
+---	you may not use this file except in compliance with the License.
+---	You may obtain a copy of the License at
+---
+---		http://www.apache.org/licenses/LICENSE-2.0
+---
+---	Unless required by applicable law or agreed to in writing, software
+---	distributed under the License is distributed on an "AS IS" BASIS,
+---	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+---	See the License for the specific language governing permissions and
+---	limitations under the License.
+----------------------------------------------------------------------------------
+
+---@type TRP3_API
+local _, TRP3_API = ...;
+local Ellyb = TRP3_API.Ellyb;
+---@type AddOn_TotalRP3
+local AddOn_TotalRP3 = AddOn_TotalRP3;
+
+local assert = assert;
+local isInstanceOf = Ellyb.Assertions.isInstanceOf;
+local after = C_Timer.After;
+local bind = Ellyb.Functions.bind;
+
+local MapScannersManager = {}
+local registeredMapScans = {};
+
+--region Total RP 3 imports
+local Events = TRP3_API.Events;
+--endregion
+
+Events.MAP_SCAN_STARTED = "MAP_SCAN_STARTED"
+Events.MAP_SCAN_ENDED = "MAP_SCAN_ENDED"
+
+
+---@param scan MapScanner
+function MapScannersManager.registerScan(scan)
```

Comment:
This'll be a contentious one; drop the superfluous "scan" suffix from the functions?

`MapScannersManager.register`, `MapScannersManager.getByID` and `MapScannersManager.launch`, etc. avoids the stutter of having "scan" mentioned twice in the name; the submodule itself already has a single responsibility so the stutter adds little clarity.

I'd go a step further and drop "Manager" too but there's existing code that follows that rule.

Reply:
I agree with you and I like that change :)